app/main.py

```
@@ -96,10 +144,360 @@ def _tc_get_json(path, params=None, timeout=30):
+def enumerate_candidate_configs(meta_runner_ids):
+    """All configs in the PARENT_PROJECT_ID subtree (excluding archived projects and
+    EXCLUDE_PROJECT_IDS) that have an ENABLED step whose runType (step.type) is one of the
+    monitored meta-runner ids. Keyed by build config id.
+    """
+    meta_set = set(meta_runner_ids)
+    excluded = set(EXCLUDE_PROJECT_IDS)
+    # Skip configs under archived projects (consistent with the other metrics). Fall back to no
+    # archived-filtering on error rather than failing the whole cycle.
+    try:
+        archived = set(get_archived_projects())
+    except Exception as e:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)